internal/uuid/uuid.go

```diff
@@ -20,3 +27,80 @@ func New(r io.Reader) (string, error) {
 	bs[6] = bs[6]&^0xf0 | 0x40
 	return fmt.Sprintf("%x-%x-%x-%x-%x", bs[0:4], bs[4:6], bs[6:8], bs[8:10], bs[10:]), nil
 }
+
+// Parse will use the google/uuid library to parse the string into a uuid
+// if parsing fails, it will return an empty map. It will fill the map
+// with some decoded values with fillMap
+func Parse(s string) (map[string]interface{}, error) {
+	uuid, err := uuid.Parse(s)
+	if err != nil {
+		return nil, err
+	}
+	out := make(map[string]interface{}, getVersionLen(int(uuid.Version())))
+	fillMap(out, uuid)
+	return out, nil
+}
+
+// Fills the map with values from the uuid. Version and variant for every version.
+// Version 1-2 has decodable values that could be of use, version 4 is random,
+// and version 3,5 is not feasible to extract data.
```

Comment:
It would be helpful if we can add some references, RFC link etc. for this explanation.

Reply:
added a link to rfc4122 that describes how they are made and the different versions of UUID